internal/snapshot/download.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package snapshot
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/flow-cli/internal/command"
+	"github.com/onflow/flow-cli/pkg/flowkit"
+	"github.com/onflow/flow-cli/pkg/flowkit/services"
+)
+
+type flagsDownload struct {
+	OutputFileName string `default:"root-protocol-state-snapshot.json" flag:"output-filename" info:"The name for the JSON output file"`
+	OutputDir      string `default:"./" flag:"output-dir" info:"The directory to output the protocol state snapshot"`
+}
+
+var downloadFlags = flagsDownload{}
+
+var DownloadCommand = &command.Command{
+	Cmd: &cobra.Command{
+		Use:     "download",
+		Short:   "Get the latest finalized protocol snapshot",
+		Example: "flow snapshot download --output-dir /tmp/snapshot",
+	},
+	Flags: &downloadFlags,
+	Run:   download,
+}
+
+func download(
+	_ []string,
+	readerWriter flowkit.ReaderWriter,
+	_ command.GlobalFlags,
+	services *services.Services,
+) (command.Result, error) {
+	snapshotBytes, err := services.Snapshot.GetLatestProtocolStateSnapshot()
+	if err != nil {
+		return nil, err
+	}
+
+	outputPath, err := filepath.Abs(filepath.Join(downloadFlags.OutputDir, downloadFlags.OutputFileName))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get absolute output path of protocol snapshot")
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to get absolute output path for protocol snapshot")
```
Though I think this code might change according to Gregor's comment https://github.com/onflow/flow-cli/pull/440/files#r789478233.

Reply:
https://github.com/onflow/flow-cli/pull/440/commits/ccb36332b51373cb03d30ad7564f670f72639391